File: Python/Problem_022.py
def name_value(name: str) -> int:
    return sum(ord(ch) - ord('A') + 1 for ch in name)

def solve_from_file(path: str = "names.txt") -> int:
    with open(path, "r", encoding="utf-8") as f:
        content = f.read().strip()

    names = [s.strip('"') for s in content.split(",")]
    names.sort()

    total = 0
    for i, name in enumerate(names, start=1):
        total += i * name_value(name)

    return total

if __name__ == "__main__":
    print(solve_from_file())  # 871198282

File: Java/Problem_022.java
import java.nio.file.*;
import java.util.*;

public class Main {

    static int nameValue(String name) {
        int sum = 0;
        for (int i = 0; i < name.length(); i++) {
            sum += name.charAt(i) - 'A' + 1;
        }
        return sum;
    }

    public static void main(String[] args) throws Exception {
        String content = Files.readString(Path.of("names.txt")).trim();
        String[] raw = content.split(",");

        ArrayList<String> names = new ArrayList<>(raw.length);
        for (String s : raw) {
            names.add(s.replace("\"", ""));
        }

        Collections.sort(names);

        long total = 0;
        for (int i = 0; i < names.size(); i++) {
            total += (long)(i + 1) * nameValue(names.get(i));
        }

        System.out.println(total); // 871198282
    }
}
